lib/clientv3/bgpconfig.go

```diff
@@ -153,3 +170,19 @@ func (r bgpConfigurations) ValidateDefaultOnlyFields(res *apiv3.BGPConfiguration
 
 	return nil
 }
+
+// Currently the name of a configuration resource has a strict format.  It is either "default"
+// for the global default values, or "node.<nodename>" for the node specific values.  Returns an
+// error if the name is in neither format.
```

Comment:
Why is this being added here and now?  If we think there's a requirement for this, would it be better to do in a separate PR that also applies the same validation to FelixConfiguration?

Reply:
This was an existing check, I copied it from updateprocessor as we will not go through that flow for BGPConfig anymore. https://github.com/projectcalico/libcalico-go/blob/master/lib/backend/syncersv1/updateprocessors/configurationprocessor.go#L332
It doesn't fit in anywhere in libcalico-go apart from clientv3/bgpconfig or we can move it to confd, log an error if the resource name doesn't match this check and skip processing that resource like we do now. I'll move this check to confd so existing behaviour doesn't change.

Reply:
Thanks for explaining; I think I vote for confd.